http/rest-client-reactive/src/test/java/io/quarkus/ts/http/restclient/reactive/UrlOverrideClientIT.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class UrlOverrideClientIT {
```

Comment:
1. What if we use non-existent url for override? How good/terrible is the error?
2. Does app writes overridden URL into log?

Reply:
1. Non-existing URL causes Exception on the service. It is possible to catch it and pretend it is OK, but that might cover some actual failures going on.
2. No it does not not log rewriting URL.

Reply:
1. I thought, that maybe we should check that error, and that it really happens, to have some non-happy-path tests. Do not know, whether it is a good idea, but please think about it.
2. And what about tracing? Is there any way for an observer to distinguish these calls?

Reply:
1. I'm not against having tests for non-happy-path. In this case, redirecting it to non-existing URL. But what would we test exactly? That something, that AFAIK cannot do anything else than fail, actually fails? Should we check that exception thrown is actually `HostNotFoundException` or something similar?  I don't see much value in that.
2. I didn't try tracing. Possibly it would work. But IMHO the module we adding it to `rest-client-reactive` is already complicated enough and I'm not sure what consequences might it have, to adding tracing to it.

Reply:
I agree, the tracing test is relevant here. Let's get OTel involved. I have vague memory that I saw some related Quarkus issue, but maybe I just remember it wrong.

Reply:
If OTel shouldn't be here, let's check if we already have a module with OTel that can be used.

Reply:
OK I agree that tracing test would be a good thing to have here. But since this looks like a big change, I will postpone it until I have time for it.

Reply:
I've added OTEL to the entire module and disabled it in default application.properties. For my local testing that caused no trouble for other tests. Also I've added testing the OTEL traces and test for failing the redirection on unknown host.